File: php/lib/steam/community/GameLeaderboardEntry.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2011, Nicholas Hastings
 *
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 */

/**
 * The GameLeaderboard class represents a leaderboard entry in a specific leaderboard
 *
 * @author     Nicholas Hastings
 * @package    steam-condenser
 * @subpackage community
 */
class GameLeaderboardEntry {

    /**
     * @var string
     */
    protected $steamId;

    /**
     * @var int
     */
    protected $score;

    /**
     * @var int
     */
    protected $rank;

    /**
     * @var GameLeaderboard
     */
    protected $leaderboard;

    /**
     * Creates a GameLeaderboardEntry object
     * @param SimpleXMLElement $entryData The entry data from Steam Community
     * @param GameLeaderboard $leaderboard The parent leaderboard that this entry belongs to
     */
    public function __construct(SimpleXMLElement $entryData, GameLeaderboard $leaderboard) {
        $this->steamId     = SteamId::create((string) $entryData->steamid, false);
        $this->score       = (int)    $entryData->score;
        $this->rank        = (int)    $entryData->rank;
        $this->leaderboard = $leaderboard;
    }

    /**
     * @return SteamId
     */
    public function getSteamId() {
        return $this->steamId;
    }

    /**
     * @return int
     */
    public function getScore() {
        return $this->score;
    }

    /**
     * @return int
     */
    public function getRank() {
        return $this->rank;
    }

    /**
     * @return GameLeaderboard
     */
    public function getLeaderboard() {
        return $this->leaderboard;
    }
}
?>

File: ruby/lib/steam/community/game_leaderboard_entry.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2011, Sebastian Staudt

# The GameLeaderboard class represents a single entry in a leaderboard
#
# @author Sebastian Staudt
class GameLeaderboardEntry

  # Returns the Steam ID of this entry's player
  #
  # @return [SteamId] The Steam ID of the player
  attr_reader :steam_id

  # Returns the score of this entry
  #
  # @return [Fixnum] The score of this player
  attr_reader :score

  # Returns the rank where this entry is listed in the leaderboard
  #
  # @return [Fixnum] The rank of this entry
  attr_reader :rank

  # Returns the leaderboard this entry belongs to
  #
  # @return [GameLeaderboard] The leaderboard of this entry
  attr_reader :leaderboard

  # Creates new entry instance for the given XML data and leaderboard
  #
  # @param [REXML::Element] entry_data The XML data of the leaderboard of the
  #        leaderboard entry
  # @param [GameLeaderboard] leaderboard The leaderboard this entry belongs to
  def initialize(entry_data, leaderboard)
    @steam_id    = SteamId.new entry_data.elements['steamid'].text.to_i, false
    @score       = entry_data.elements['score'].text.to_i
    @rank        = entry_data.elements['rank'].text.to_i
    @leaderboard = leaderboard
  end

end
